File: problems/35-search-insert-position.py
class Solution:
    def searchInsert(self, nums, target):
        left = 0
        right = len(nums) - 1
        while left <= right:
            mid = left + (right - left) // 2
            mid_value = nums[mid]
            if mid_value >= target:
                right = mid - 1
            else:
                left = mid + 1
        return left

File: leetcode/java/35SearchInsertPosition.java
/**
 * T: O(logN) S: O(1)
 *
 * <p>Peform a binary search.
 */
class Solution {
  public int searchInsert(int[] nums, int target) {
    int lo = 0;
    int hi = nums.length - 1;
    while (lo <= hi) {
      int mid = lo + (hi - lo) / 2;
      int atMid = nums[mid];
      if (atMid == target) {
        return mid;
      }
      if (atMid > target) {
        hi = mid - 1;
      } else {
        lo = mid + 1;
      }
    }
    return lo;
  }
}
